lib/parse-route-string.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+/**
+ * Take a WP route string (with PCRE named capture groups), such as
+ * @module parseRouteString
+ */
+var util = require( 'util' );
+var _ = require( 'lodash' );
+function logFull( obj ) {
+	console.log( util.inspect( obj, {
+		colors: true,
+		depth: null
+	}) );
+}
+
+// All valid routes in API v2 beta 11
+var routes = require( './endpoint-response.json' ).routes;
+
+function forIn( obj, callbackFn ) {
+	Object.keys( obj ).forEach(function( key, idx ) {
+		callbackFn( obj[ key ], idx );
+	});
+}
+
+/*
+
+"/wp/v2/posts": {},
+"/wp/v2/posts/(?P<id>[\\d]+)": {},
+"/wp/v2/posts/(?P<parent>[\\d]+)/revisions": {},
+"/wp/v2/posts/(?P<parent>[\\d]+)/revisions/(?P<id>[\\d]+)": {},
+"/wp/v2/pages": {},
+"/wp/v2/pages/(?P<id>[\\d]+)": {},
+"/wp/v2/pages/(?P<parent>[\\d]+)/revisions": {},
+"/wp/v2/pages/(?P<parent>[\\d]+)/revisions/(?P<id>[\\d]+)": {},
+
+ */
+
+var buildRouteTree = require( './build-route-tree' );
+var routesByNamespace = buildRouteTree( routes );
+
+logFull( routesByNamespace );
+
+/*
+
+I want to deduce the following API from this tree (or one like it):
+
+wp.posts();                     /wp/v2/posts
+wp.posts().id( 7 );             /wp/v2/posts/7
+wp.posts().id().revisions();    /wp/v2/posts/7/revisions
+wp.posts().id().revisions( 8 ); /wp/v2/posts/7/revisions/8
```

Comment:
This is an example of the existing API, which I hope to derive from the tree representation above